### PR TITLE
Deprecate `script::read_uint`

### PR DESCRIPTION
**Pull request by tcharding**
Patch one does the deprecation, the rest of the PR is made up of tiny improvements to the code around reading/writing 'scriptint's (conceptually `CScriptNum`s). I did all this while trying to decipher the discussion on #1547. 

### Note Please

There are many more changes in the pipeline for all this read/write "script int" stuff. This PR was done ages ago and I believe it stall adds value.

I re-did the whole PR manually because of the recent `script` module changes. I hope no one else has to do that - if you do please feel free to holla and I'll "rebase" your PR for you.

### CODE REVIEW

**Comment by tcharding** [comment]
Note to merger, may need to close #1547 manually .

**Comment by tcharding** [comment]
Damn, looks like we have to solve this ASAN problem.

**Comment by apoelstra** [comment]
Can you rebase since ASAN is fixed?

**Comment by tcharding** [comment]
Rebased to pick up CI fix, no changes.

**Review by apoelstra — APPROVED** [review]
ACK 36ed20ddaf5ce458097ee85b2e0a30a547236e28

**Comment by tcharding** [comment]
"rebased" manually (re-did the whole thing) and re-wrote PR description.

File: bitcoin/src/blockdata/script/mod.rs [hunk 1]
```diff
@@ -167,6 +167,7 @@ pub fn read_scriptbool(v: &[u8]) -> bool {
 /// Note that this does **not** return an error for `size` between `core::size_of::<usize>()`
 /// and `u16::max_value / 8` if there's no overflow.
 #[inline]
+#[deprecated(since = "0.30.0", note = "bitcoin integers are signed 32 bits, use read_scriptint")]
```
Comment (Kixunil) [inline bitcoin/src/blockdata/script/mod.rs:177]: IIUC this was meant only for reading push data length and got published by accident.

File: bitcoin/src/blockdata/script/mod.rs [hunk 2]
```diff
@@ -115,6 +115,8 @@ pub fn write_scriptint(out: &mut [u8; 8], n: i64) -> usize {
 /// don't fit in 64 bits (for efficiency on modern processors) so we
 /// simply say, anything in excess of 32 bits is no longer a number.
 /// This is basically a ranged type implementation.
+///
+/// This code is based on the `CScriptNum` constructor in Bitcoin Core (see `script.h`).
```
Comment (Kixunil) [inline bitcoin/src/blockdata/script/mod.rs:119]: That's an implementation detail. I wouldn't use doc comment.

**Review by Kixunil — APPROVED** [review]
ACK a7117bf8f1ca295258c25c7d2eb5646bf78da078

File: bitcoin/src/blockdata/script/mod.rs (hunk 2 above)
Comment (apoelstra) [reply bitcoin/src/blockdata/script/mod.rs:119]: Agreed, but I'm not going to block the PR on it.

**Review by apoelstra — APPROVED** [review]
ACK a7117bf8f1ca295258c25c7d2eb5646bf78da078

**Pull request merged**